# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/ConnectionEntry.cpp
# Commit: 373bc48d5325
# Full Hash: 373bc48d5325e17cb67a11035d4542d221023594
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2023-09-19 09:37:28
# Regressor Bug: 1784261
# File Overlap Count: 1
# Description:
#   Bug 1784261 - Add telemetry to understand the reason of connection closure, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188132
# ==============================================================================

diff -r f2fee3884997 -r 373bc48d5325 netwerk/protocol/http/ConnectionEntry.cpp
--- a/netwerk/protocol/http/ConnectionEntry.cpp	Mon Sep 18 14:28:30 2023 +0000
+++ b/netwerk/protocol/http/ConnectionEntry.cpp	Mon Sep 18 14:44:18 2023 +0000
@@ -544,6 +544,8 @@
            "because new "
            "spdy connection (%p) takes precedence\n",
            otherConn, conn));
+      otherConn->SetCloseReason(
+          ConnectionCloseReason::CLOSE_EXISTING_CONN_FOR_COALESCING);
       otherConn->DontReuse();
     }
   }